develop/items/custom-enchantment-effects.md

```diff
@@ -0,0 +1,97 @@
+
+```java
+@Override
+public void apply(ServerWorld world, int level, EnchantmentEffectContext context, Entity target, Vec3d pos) {
+    if (target instanceof LivingEntity victim){
+        if (context.owner() != null) {
+            // Spawn lightning strikes at the target based on enchantment level
+            float numStrikes = this.amount.getValue(level);
+            for (float i = 0; i < numStrikes; i++) {
+                BlockPos position = victim.getBlockPos();
+                EntityType.LIGHTNING_BOLT.spawn(world, position, SpawnReason.TRIGGERED);
+            }
+        }
+    }
+}
+```
+
+## Registering the Enchantment Effect {#registering-the-enchantment-effect}
+
+Like every other component of your mod, we'll have to add this EnchantmentEffect to Minecraft's registry. To do so, add a class `ModEnchantmentEffects` (or whatever you want to name it) and a helper method to register the enchantment. Be sure to call the `registerModEnchantmentEffects()` in your main class (FabricDocsReference.java if you are following the docs).
+
+@[code transcludeWith=#entrypoint](@/reference/latest/src/main/java/com/example/docs/enchantment/ModEnchantmentEffects.java)
+
+
+## Creating the Enchantment {#creating-the-enchantment}
+
+Now we have an enchantment effect! Lastly, we'll create an enchantment to apply our custom effect to. This can be done by creating a JSON file like in datapacks, however, in this guide we'll generate the JSON dynamically using Fabric's data generation tools. To start, create a folder `data` and within it create an `EnchantmentGenerator` class; optionally you can keep this class within your `enchantment` folder.
+
+Within this class, we'll first register a new enchantment, and then use the `configure()` method to create our JSON programatically.
+
+@[code transcludeWith=#entrypoint](@\reference\latest\src\main\java\com\example\docs\data\EnchantmentGenerator.java)
+
+Lastly, we must tell our mod to add our `EnchantmentGenerator` to the list of data generation tasks. To do so, ensure your project has data generation turned on/configured correctly, and look for (or create if it doesn't exit) a data generation class; this will usually be at the root of your package (same folder as your main class). Lastly, add the `EnchantmentGenerator` to this inside of the `onInitializeDataGenerator` class.
+
+```java
+public void onInitializeDataGenerator(FabricDataGenerator generator) {
+		FabricDataGenerator.Pack pack = generator.createPack();
+		pack.addProvider(EnchantmentGenerator::new);
+	}
+```
+
+
+
+The code above will automatically generate the respective JSON files for your enchantment when you run the `Data Generation` task in your gradle. This task only needs to be run once when you update this file. 
```

Comment:
Not needed, leave this to the data generation page. 

Instead maybe something like 

"Now, when you run your mod's data generation task, enchantment JSONs will be generated. An example can be seen below: ... [code reference here to JSON file generated]"